src/AI/Connection.php

```diff
@@ -46,6 +50,51 @@ public function fetch_ai_response( $token, $prompt, $timeout = 15 ) {
 		return json_decode( $body, true );
 	}
 
+	/**
+	 * Fetch the AI responses in parallel using the given token and prompts.
+	 *
+	 * @param string $token The JWT token.
+	 * @param array  $prompts The prompts to send to the API.
+	 * @param int    $timeout The timeout for the request.
+	 *
+	 * @return array|WP_Error The responses or a WP_Error object.
+	 */
+	public function fetch_ai_responses( $token, array $prompts, $timeout = 15 ) {
+		if ( $token instanceof \WP_Error ) {
+			return $token;
+		}
+
+		$requests = array();
+		foreach ( $prompts as $prompt ) {
+			$requests[] = array(
+				'url'     => self::TEXT_COMPLETION_API_URL,
+				'type'    => 'POST',
+				'headers' => array( 'Content-Type' => 'application/json; charset=utf-8' ),
+				'data'    => wp_json_encode(
+					array(
+						'feature' => 'woocommerce_blocks_patterns',
+						'prompt'  => $prompt,
+						'token'   => $token,
+					)
+				),
+			);
+		}
+
+		$responses = Requests::request_multiple( $requests, array( 'timeout' => $timeout ) );
```

Comment:
It is a great idea to make the requests run in parallel: this should save us a lot of time during the onboarding :)